web/modules/custom/contact_form/src/EventSubscriber/NotificationSubscriber.php

```diff
@@ -76,19 +63,160 @@ public function contactForm(NotificationEvent $event) {
   }
 
   /**
-   * Returns the users with a given role.
+   * Determines the entity object of the route matching a uri.
    *
-   * @param string $role_id
-   *   The role id.
+   * @param string $uri
+   *   The internal or external uri.
+   *
+   * @return \Drupal\Core\Entity\EntityInterface|null
+   *   The entity object or null in any other case.
+   */
+  protected function getEntityFromUrl($uri) {
+    try {
+      $url = Url::fromUri($uri);
+    }
+    catch (Exception $e) {
+      return NULL;
+    }
+
+    if (!$url->isRouted()) {
+      return NULL;
+    }
+
+    $route_parameters = $url->getRouteParameters();
+    if (!isset($route_parameters['node']) && !isset($route_parameters['rdf_entity'])) {
+      return NULL;
+    }
+
+    // Check what type of content we are reporting.
+    $entity_type_id = NULL;
+    if (isset($route_parameters['node'])) {
```

Comment:
Well, this entity will be used to get memberships and a title. It does not matter which revision is reported as if it's an earlier version, it's not meant to be seen by everyone. 
Also, in the same method, there is a check about the canonical path. so in case of the node revision link, it will not be valid anyway.